client/src/App/App.tsx

```diff
@@ -36,6 +37,45 @@ import './App.css';
 
 const { Content } = Layout;
 
+interface ComponentProps extends Partial<RouteProps> {
+  component: any;
+  redirectPath: string;
+  disableLoginProtection: boolean;
+  path: string;
+}
+
+export const PrivateComponent = ({ component: Component, ...rest }: ComponentProps) => {
+  return (
+    <ConnectedPrivateRoute
+      {...rest}
+      component={(props: RouteComponentProps) => (
+        <Component {...props} keycloakBaseURL={KEYCLOAK_API_BASE_URL}/>
+      )}
+    />
+  );
+};
+
+export const PublicComponent = ({ component: Component, ...rest}: Partial<ComponentProps>) => {
+  return (<Route {...rest} component={(props: RouteComponentProps) => <Component {...props}/> }/>);
+};
+
+export const CallbackComponent = (routeProps: RouteComponentProps<RouteParams>) => {
+  if (BACKEND_ACTIVE) {
+    return <CustomConnectedAPICallBack {...routeProps} />;
+  }
+  return (
+    <ConnectedOauthCallback
+      SuccessfulLoginComponent={() => {
+        return <Redirect to="/" />;
+      }}
+      LoadingComponent={Loading}
+      providers={providers}
+      oAuthUserInfoGetter={getOpenSRPUserInfo}
+      {...routeProps}
+    />
+  );
+};
```

Comment:
it would be a good idea to add a docstring on how these components are/should be
 used.

Reply:
addressed https://github.com/OpenSRP/web/pull/116/commits/cf526007a779ba76afd29b80198d21228e32fddc